### ISSUE TITLE
Calculating the Fractional Occupancies for Each Segment / Session

### ISSUE THREAD

**Issue by Bingram22** [OHBA-analysis/HMM-MAR#63]
Hi Diego, 
I am currently working on some eyes open / eyes closed resting-state EEG data and I would like to calculate the fractional occupancies (as well as lifetimes etc) for each condition/session. There is a total of 18 participants, each with four sessions [eyes open, eyes closed, eyes open, eyes closed]. I have trained the HMM, however, when I try to obtain the fractional occupancies, I can only calculate the FO per participant (e.g. participants x states). 

I was wondering if there is currently a way that would allow me to calculate it per session (e.g. sessions x states)? 

Many thanks!
Brandon

**Comment by jnvandermeer** [comment]
Hi Brandon,

I have the exact same situation - EO/EC EEG data. In the end, what I did was to make a function that takes the whole concatenated data (mat, gamma and T), and based on specific conditions spits out a new mat, gamma and T, containing only data from those conditions/subjects. 

There is also a means of specifying which samples you want to calculate the FO over, built into the HMM functions themselves that calculate FO etc.

If you annotate EEG and gotten rid of bad segments, or wish to plot the paths, then this comes with a fair bit of bookkeeping. In my case I made new data structures that annotate each segment (T) with some extra info so I later don't have to think it anymore.

**Comment by Bingram22** [comment]
Hi Johan, 
Thanks for your reply! It's reassuring to know someone else is using this method on a similar dataset. I think I finally figured out it using the HMM functions. 

I think the issue was that I had defined T as:

`{[subj1_eo, subj1_ec], [subj2_eo, subj2_ec], ... [subjn_eo, subjn_ec]}`

 for the purpose of training the model since I was using stochastic inference.

 Defining T as one long vector just with all the durations for each session has given me the FO for each session / segment (i think). e.g.:

`[subj1_eo, subj1_ec, subj2_eo, subj2_ec, ... subjn_eo,subjn_ec] `


Many thanks again for your help!
Brandon
